Sources/muterCore/CLICommands/RunCommand/RunCommandHandler.swift

```diff
@@ -9,24 +11,16 @@ class RunCommandHandler {
         self.state = state
     }
     
-    init(options: RunCommandOptions,
+    init(command: Run,
          steps: [RunCommandStep] = RunCommandHandler.defaultSteps) {
         self.steps = steps
-        self.state = RunCommandState(from: options)
+        self.state = RunCommandState(from: command)
     }
     
-    func handle() -> Result<(), MuterError> {
-        for step in steps {
-            let result = step.run(with: state)
-
-            switch result {
-            case .failure(let error):
-                return .failure(error)
-            case .success(let stateChanges):
-                state.apply(stateChanges)
-            }
+    func handle() throws {
+        try steps.forEach {
+            try $0.run(with: state).map(state.apply(_:)).get()
```

Comment:
I wrote this line, but I didn't love it. I think it's a little too cute/compact. Feel free to make it more readable if you have ideas for how to do that.